### ISSUE TITLE
Can't remove apiVersion from helm chart

### ISSUE THREAD

**Issue by nesl247** [pulumi/pulumi-kubernetes#1034]
### Problem description

In https://github.com/Kong/charts/blob/master/charts/kong/templates/admission-webhook.yaml, they do

```
{{- if .Capabilities.APIVersions.Has "admissionregistration.k8s.io/v1" }}
apiVersion: admissionregistration.k8s.io/v1
{{- else }}
apiVersion: admissionregistration.k8s.io/v1beta1
{{- end }}
```

at: https://github.com/Kong/charts/blob/dec24148f49f36bd4902142ce1a8360d2c681aa2/charts/kong/templates/admission-webhook.yaml#L6-L10

I haven't seen a way to specify that only a specific version of an API is available. The `apiVersions` feature seems to add to the list, not set the specific version used.

### Reproducing the issue

```
new k8s.helm.v3.Chart(`${config.projectName}-internal`, {
  chart: 'kong',
  version: config.project.require('chartVersion'),
  fetchOpts: {
    repo: 'https://charts.konghq.com'
  },
  namespace: namespace.metadata.name,
  apiVersions: ['admissionregistration.k8s.io/v1beta1'],
  values: {
    ...sharedValues,
    installCRDs: true,
    fullName: 'kong-internal',
    proxy: {
      annotations: {
        'cloud.google.com/load-balancer-type': 'Internal'
      }
    }
  }
});
```

#196 added the `apiVersions` feature, but given the lack of documentation in both `Pulumi` and `helm` regarding this, I'm not sure if I'm doing something wrong or not.

**Comment by lblackstone** [comment]
This appears to be a bug in Helm v3, and isn't specific to Pulumi. I confirmed that the Chart renders as expected with the `--api-versions` flag on Helm v2, but does not on v3.

Related: https://github.com/databus23/helm-diff/issues/165

As a workaround, you should be able to use the [transformations option](https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/kubernetes/helm/v2/#ChartOpts-transformations) to fix the rendered resource.

**Comment by nesl247** [comment]
From what I can see, this was patched in helm 3.1.0. I'm on 3.1.2 and still having the same issue. https://github.com/databus23/helm-diff/commit/62419ae61426ac1e2da9d27aeeb3b2d0d93956f6 was the patch for this. It mentions `--upgrade` and such, but that isn't what we are using here. It looks to me like pulumi needs to change how it calls helm to make this work.
